pkg/rbac/middleware.go

```diff
@@ -0,0 +1,97 @@
+package rbac
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/stolostron/search-v2-api/pkg/config"
+	authv1 "k8s.io/api/authentication/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+)
+
+var kClientset *kubernetes.Clientset
+
+func KubeClient() {
+	config, err := config.GetClientConfig()
+	if err != nil {
+		klog.Fatal(err.Error())
+	}
+	kClientset, err = kubernetes.NewForConfig(config)
+
+	if err != nil {
+		klog.Fatal(err.Error())
+	}
+
+}
+
+//verifies token (userid) with the TokenReview:
+func Middleware() func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			//if there is cookie available use that else use the authorization header:
+			var clientToken string
+			if len(r.Cookies()) > 0 {
+				for _, cookie := range r.Cookies() {
+					if cookie.Name == "acm-access-token-cookie" {
```

Comment:
You can access the cookie by name instead of iterating through all cookies.
https://pkg.go.dev/net/http#Request.Cookie

```go
cookie, err := r.Cookie("acm-access-token-cookie")
```